### PR TITLE
documentation added for pipelines

### PR DESCRIPTION
**Pull request by enekofb**
Closes https://github.com/weaveworks/weave-gitops-enterprise/issues/1274

**What changed?**
Added documentation for pipelines

**Why was this change made?**

In order to document the released feature

**How was this change implemented?**

Adding documentation. It is place under the enterprise section, differently to other enterprise features in the main documentation root. the motivation is that seems to be a more clear navigation for users. 

**How did you validate the change?**

Locally 
Screenshot 2022-10-03 at 11 07 57

**Release notes**

- No

**Documentation Changes**
- It is a documentation change itself

### CODE REVIEW

File: website/docs/enterprise/pipelines/getting-started.mdx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+import TierLabel from "../../_components/TierLabel";
+
+import CodeBlock from "@theme/CodeBlock";
+import BrowserOnly from "@docusaurus/BrowserOnly";
+
+# Getting started with pipelines <TierLabel tiers="enterprise" />
+
+## Requirements
+- You have configured Weave Gitops Enterprise [RBAC for pipelines](../authorization)
```
Comment (yiannistri) [inline website/docs/enterprise/pipelines/getting-started.mdx:15]:
```suggestion
- You have configured Weave GitOps Enterprise [RBAC for pipelines](../authorization)
```

File: website/docs/enterprise/pipelines/getting-started.mdx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+## Define a pipeline
+
+A pipeline allows you to define the route your application is taking in order to make it to production.
+There are three main concepts playing in a pipeline:
+- the `application` to deliver
+- the `environments` that your app will go through in its way to production (general)
+- the `deployment targets` or the clusters that each environment has
+
+In weave gitops you could define a delivery pipeline using a `Pipeline` Custom Resource Definitions.
```
Comment (yiannistri) [inline website/docs/enterprise/pipelines/getting-started.mdx:25]:
```suggestion
You can define a delivery pipeline using a `Pipeline` custom resource.
```

File: website/docs/enterprise/pipelines/getting-started.mdx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    - name: prod
+      targets:
+        - namespace: podinfo-02-prod
+          clusterRef:
+            kind: GitopsCluster
+            name: prod
+            namespace: flux-system
+```
+
+In the previous example, `podinfo` application is delivered to a traditional pipeline composed by `dev`, `qa`, (`perf`) and `production` environments.
```
Comment (yiannistri) [inline website/docs/enterprise/pipelines/getting-started.mdx:68]: Not sure why `perf` needs to be wrapped in round brackets.

File: website/docs/enterprise/pipelines/getting-started.mdx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+Each environment is backed by a `GitopsCluster` [deployment target](../../../cluster-management/managing-existing-clusters/).
+
+For more details about the spec of a pipeline [see here](spec/v1alpha1/pipeline.mdx)
+
+## View pipeline list
+
+Once flux has created your pipeline you could navigate to the pipelines view to see it.
```
Comment (yiannistri) [inline website/docs/enterprise/pipelines/getting-started.mdx:75]:
```suggestion
Once flux has created your pipeline you can navigate to the pipelines view to see it.
```

File: website/docs/enterprise/pipelines/getting-started.mdx [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+![view pipelines](img/view-pipelines.png)
+
+Pipeline list view show the list of pipeline you have access to. For each pipeline, a simplified view of the pipeline
```
Comment (yiannistri) [inline website/docs/enterprise/pipelines/getting-started.mdx:79]:
```suggestion
Pipeline list view show the list of pipelines you have access to. For each pipeline, a simplified view of the pipeline
```

File: website/docs/enterprise/pipelines/getting-started.mdx [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+is shown with the application `Type` and `Environments` it goes through.
+
+## View a pipeline details
+
+Once you have selected a pipeline from the list, you will navigate to its details view.
+In pipeline details view you could view the current status of your application by environment and deployment
```
Comment (yiannistri) [inline website/docs/enterprise/pipelines/getting-started.mdx:85]:
```suggestion
In pipeline details view you can view the current status of your application by environment and deployment
```

File: website/docs/enterprise/pipelines/intro.mdx [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+## Pipelines
+
+As [wikipedia defines](https://en.wikipedia.org/wiki/Continuous_delivery), Continuous delivery is
+
+>a software engineering approach in which teams produce software in short cycles,
+>ensuring that the software can be reliably released at any time and, when releasing the software, without doing so manually.
+>It aims at building, testing, and releasing software with greater speed and frequency.
+
+>Continuous delivery is enabled through the deployment pipeline.
+>The purpose of the deployment pipeline has three components: visibility, feedback, and continually deploy.[12]
```
Comment (yiannistri) [inline website/docs/enterprise/pipelines/intro.mdx:22]:
```suggestion
>The purpose of the deployment pipeline has three components: visibility, feedback, and continually deploy.
```

File: website/docs/enterprise/pipelines/intro.mdx [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+Weave Gitops Enterprise Pipelines allows you to define your deployment pipelines to enable continuous delivery for
```
Comment (yiannistri) [inline website/docs/enterprise/pipelines/intro.mdx:24]:
```suggestion
Weave GitOps Enterprise Pipelines allows you to define your deployment pipelines to enable continuous delivery for
```

File: website/docs/enterprise/pipelines/intro.mdx [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+your gitops applications.
+
+As part of Weave Gitops Enterprise, you could use
```
Comment (yiannistri) [inline website/docs/enterprise/pipelines/intro.mdx:27]:
```suggestion
As part of Weave GitOps Enterprise, you can
```

File: website/docs/enterprise/pipelines/intro.mdx [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+
+- [Define a delivery pipeline](../getting-started/#define-a-pipeline) that you want you application to follow
+- [View a pipeline](../getting-started/#view-pipeline-list) and check the status of your deployments.
+
+Now that you know what delivery pipelines could do for you, just [get started](../getting-started)
```
Comment (yiannistri) [inline website/docs/enterprise/pipelines/intro.mdx:32]:
```suggestion
Now that you know what delivery pipelines can do for you, follow the [guide to get started](../getting-started).
```

File: website/docs/enterprise/pipelines/getting-started.mdx [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+## View pipeline list
+
+Once flux has created your pipeline you could navigate to the pipelines view to see it.
+
+![view pipelines](img/view-pipelines.png)
+
+Pipeline list view show the list of pipeline you have access to. For each pipeline, a simplified view of the pipeline
+is shown with the application `Type` and `Environments` it goes through.
+
+## View a pipeline details
```
Comment (yiannistri) [inline website/docs/enterprise/pipelines/getting-started.mdx:82]:
```suggestion
## View the details of a pipeline
```

File: website/docs/enterprise/pipelines/getting-started.mdx [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+            name: prod
+            namespace: flux-system
+```
+
+In the previous example, `podinfo` application is delivered to a traditional pipeline composed by `dev`, `qa`, (`perf`) and `production` environments.
+Each environment is backed by a `GitopsCluster` [deployment target](../../../cluster-management/managing-existing-clusters/).
+
+For more details about the spec of a pipeline [see here](spec/v1alpha1/pipeline.mdx)
+
+## View pipeline list
```
Comment (yiannistri) [inline website/docs/enterprise/pipelines/getting-started.mdx:73]:
```suggestion
## View the list of pipelines
```

File: website/docs/enterprise/pipelines/spec/v1alpha1/pipeline.mdx [hunk 13]
```diff
@@ -0,0 +1,167 @@
+---
+title: Pipeline
+sidebar_position: 1
+hide_title: true
+---
+# Pipeline <TierLabel tiers="enterprise" />
+
+The Pipeline API defines a resource for continuous delivery pipelines.
+
+An example of fully defined pipeline could be found below.
```
Comment (yiannistri) [inline website/docs/enterprise/pipelines/spec/v1alpha1/pipeline.mdx:10]:
```suggestion
An example of a fully defined pipeline is shown below.
```

**Review by yiannistri — COMMENTED** [review]
Looks good to me 👍 I'm not sure we need to show details of the Go types of the pipeline and keep that reference always current going forward if we don't provide a way to interact with it programmatically. But I can be convinced otherwise.

**Comment by enekofb** [comment]
> Looks good to me 👍 I'm not sure we need to show details of the Go types of the pipeline and keep that reference always current going forward if we don't provide a way to interact with it programmatically. But I can be convinced otherwise.

@yiannistri 

do you mean?

https://github.com/weaveworks/weave-gitops/pull/2793/files#diff-e5b3bdfc3a2cfcc6447284d76b58139d8df76bb32b72786e018f8e82cb9ef3fcR56

it just mimics flux documentation https://github.com/fluxcd/helm-controller/blob/main/docs/spec/v2beta1/helmreleases.md

mentioned in the standup, not sure whether this is the approach that we want to follow so feel free to share preference / alternative

**Comment by sympatheticmoose** [comment]
Thank you @enekofb for taking the time to write these docs. They're good - my comments are mostly nits. 

On reflection, I am wondering about the nesting under enterprise and some placement of the docs. However, we are about to revisit the overall structure and more changes may occur. In terms of expedience, please feel free to merge in their current form, and we can use some of the comments I've made in this PR when reworking the site structure.

**Comment by yiannistri** [comment]
> do you mean?
> 
> https://github.com/weaveworks/weave-gitops/pull/2793/files#diff-e5b3bdfc3a2cfcc6447284d76b58139d8df76bb32b72786e018f8e82cb9ef3fcR56
> 
> it just mimics flux documentation https://github.com/fluxcd/helm-controller/blob/main/docs/spec/v2beta1/helmreleases.md
> 
> mentioned in the standup, not sure whether this is the approach that we want to follow so feel free to share preference / alternative

Yes I was referring to those. I think they're useful in general but I would suggest postpone adding them because a) we're still making changes to the CRD for promotions and b) the current spec is not as sophisticated as the Flux primitives to justify adding them to docs.

Other than that, the docs look good so I'll defer to your judgement regarding this.

**Review by yiannistri — APPROVED** [review]

**Comment by enekofb** [comment]
> > do you mean?
> > https://github.com/weaveworks/weave-gitops/pull/2793/files#diff-e5b3bdfc3a2cfcc6447284d76b58139d8df76bb32b72786e018f8e82cb9ef3fcR56
> > it just mimics flux documentation https://github.com/fluxcd/helm-controller/blob/main/docs/spec/v2beta1/helmreleases.md
> > mentioned in the standup, not sure whether this is the approach that we want to follow so feel free to share preference / alternative
> 
> Yes I was referring to those. I think they're useful in general but I would suggest postpone adding them because a) we're still making changes to the CRD for promotions and b) the current spec is not as sophisticated as the Flux primitives to justify adding them to docs.
> 
> Other than that, the docs look good so I'll defer to your judgement regarding this.

@yiannistri sounds good, i could remove it however there is gap that I could think we need to cover which is 

as user without pipelines knowledge, i want to have some documentation to write a pipeline 

for example to understand the different namespaces or so 

https://github.com/weaveworks/weave-gitops/pull/2793/files#diff-e5b3bdfc3a2cfcc6447284d76b58139d8df76bb32b72786e018f8e82cb9ef3fcR90-R93

this is part of the CRD / generated go types. 

How should we capture this?

**Comment by yiannistri** [comment]
Fair enough 👍 Let's get this in.

File: website/docs/enterprise/pipelines/getting-started.mdx (hunk 3 above)
Comment (enekofb) [reply website/docs/enterprise/pipelines/getting-started.mdx]: remove the wrapping

**Pull request merged**